django/db/models/fields/json.py

```diff
@@ -0,0 +1,523 @@
+import json
+
+from django import forms
+from django.conf import settings
+from django.core import checks, exceptions
+from django.db import connections
+from django.db.models import Func, Value, lookups
+from django.db.models.lookups import (
+    FieldGetDbPrepValueMixin, Lookup, Transform,
+)
+from django.db.utils import NotSupportedError
+from django.utils.translation import gettext_lazy as _
+
+from . import Field, TextField
+from .mixins import CheckFieldDefaultMixin
+
+
+class JSONField(CheckFieldDefaultMixin, Field):
+    description = _('A JSON object')
+    default_error_messages = {
+        'invalid': _('Value must be valid JSON.'),
+    }
+    _default_hint = ('dict', '{}')
+
+    def __init__(self, encoder=None, decoder=None, default=dict, *args, **kwargs):
+        self.encoder, self.decoder = encoder, decoder
+        super().__init__(default=default, *args, **kwargs)
+
+    def check(self, **kwargs):
+        return [
+            *super().check(**kwargs),
+            *self._check_json_support(),
+        ]
+
+    @classmethod
+    def _check_json_support(cls):
+        errors = []
+        for db in settings.DATABASES:
+            connection = connections[db]
+            if not connection.features.supports_json_field:
+                errors.append(
+                    checks.Error(
+                        "Connection '%s' uses a version of %s that does not support JSONField." % (
+                            db, connection.display_name
+                        ),
+                        hint=(
+                            'See the documentation of JSONField for supported database versions.'
+                        ),
+                        obj=cls
+                    )
+                )
```

Comment:
This check ensures that *all* database backends support the field. Often projects add a second database in DATABASES that's used for just a few models, for example pulling some data from a reporting system. A database router class often routes queries for just those few tables there.

The checks in Django-MySQL are deliberately more lax, and only return an Error if there is *no* connection that supports the current field, for example: https://github.com/adamchainz/django-mysql/blob/master/django_mysql/models/fields/json.py#L69

I think we should adopt the same logic here :)

Reply:
Yes, I [wondered](https://github.com/django/django/pull/11452#discussion_r307073935) about that. Thanks for the justification.